### ISSUE TITLE
Decouple Crowdsale from MintableToken

### ISSUE THREAD

**Issue by frangio** [OpenZeppelin/zeppelin-solidity#351]
(Extracted verbatim from #310 by @jakub-wojciechowski.)

Crowdsale contract is tightly coupled to MintableToken. Wouldn’t it be better to make the base Crowdsale as generic as possible? It will allow others to implement a model when the token pool is created before a sale and then transferred rather than minted. Currently, it’s impossible to override methods like:

```
function createTokenContract() internal returns (MintableToken) {
```

I will suggest extracting:

```
token.mint(beneficiary, tokens);
```

to

```
function distribute(beneficiary, tokens) internal {
  token.mint(beneficiary, tokens); 
}
```

And probably rename:

```
function createTokenContract() => setupTokenContract()
```

**Comment by amazingandyyy** [comment]
I noticed the same issue when i want to switch from mintable to other token, I think the problem is `~zeppelin-solidity/contracts/crowdsale/Crowdsale.sol` this contract uses `mint` as default method of the token, cause all crowdsale can only work mintable token.

**Comment by ajsantander** [comment]
Addressed in the crowdsales refactor #744.
The base Crowdsale contract now refers to a generic ERC20 token and can be extended to implement `transferFrom`, minting, etc.
